Forta-Bots-Challenge-3/src/agent.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Finding,
+  HandleBlock,
+  getEthersProvider,
+  BlockEvent,
+  getAlerts,
+} from "forta-agent";
+import {
+  DAI_ADDRESS,
+  L1_ESCROW_ARBITRUM,
+  L1_ESCROW_FUNCTION_SIGNATURE,
+  L1_ESCROW_OPTIMISM,
+  BOT_ID,
+} from "./constants";
+import { ethers, Contract } from "ethers";
+import { getL1Finding, checkBlock } from "./utils";
+
+export function provideHandleBlock(
+  provider: ethers.providers.Provider,
+): HandleBlock {
+  return async function handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    const { chainId } = await provider.getNetwork();
+
+    try {
+      if (chainId == 1) {
+        const daiContract = new Contract(
+          DAI_ADDRESS,
+          L1_ESCROW_FUNCTION_SIGNATURE,
+          provider,
+        );
+        findings.push(
```

Comment:
Just as an optimization recommendation, you can consider firing this alert only in those blocks where the balance has changed.